server/src/main/java/org/elasticsearch/search/fetch/subphase/FieldFetcher.java

```diff
@@ -148,6 +164,17 @@ private FieldFetcher(
         if (this.unmappedFieldsFetchAutomaton != null) {
             collectUnmapped(documentFields, sourceLookup.source(), "", 0);
         }
+        if (this.unmappedConcreteFields != null) {
```

Comment:
Small comment, rereading this maybe it makes sense to move this into `collectUnmapped`? That way all the unmapped logic is consolidated into one method.

Reply:
Done.